pkg/auth/providers/common/ldap/ldap_util.go

```diff
@@ -211,6 +226,75 @@ func AttributesToPrincipal(attribs []*ldapv2.EntryAttribute, dnStr, scope, provi
 	return principal, nil
 }
 
+func GatherParentGroups(groupPrincipal v3.Principal, searchDomain string, groupScope string, config *ConfigAttributes, lConn *ldapv2.Conn,
+	groupMap map[string]bool, nestedGroupPrincipals *[]v3.Principal) error {
+	groupMap[groupPrincipal.ObjectMeta.Name] = true
+	principals := []v3.Principal{}
+	var searchAttributes []string
+	parts := strings.SplitN(groupPrincipal.ObjectMeta.Name, ":", 2)
+	if len(parts) != 2 {
+		return errors.Errorf("invalid id %v", groupPrincipal.ObjectMeta.Name)
+	}
+	groupDN := strings.TrimPrefix(parts[1], "//")
+
+	if config.ProviderName == "activedirectory" {
+		searchAttributes = []string{config.MemberOfAttribute, config.ObjectClass, config.GroupObjectClass, config.UserLoginAttribute, config.GroupNameAttribute,
+			config.GroupSearchAttribute}
+	} else if config.ProviderName == "openldap" {
+		searchAttributes = []string{config.GroupMemberUserAttribute, config.GroupMemberMappingAttribute, config.ObjectClass, config.GroupObjectClass, config.UserLoginAttribute,
+			config.GroupNameAttribute, config.GroupSearchAttribute}
+	}
+	searchGroup := ldapv2.NewSearchRequest(searchDomain,
+		ldapv2.ScopeWholeSubtree, ldapv2.NeverDerefAliases, 0, 0, false,
+		fmt.Sprintf("(&(%v=%v)(%v=%v))", config.GroupMemberMappingAttribute, ldapv2.EscapeFilter(groupDN), config.ObjectClass, config.GroupObjectClass),
+		searchAttributes, nil)
```

Comment:
This query tries to find all groups which have this particular group mentioned in its 'member' list. This has been the OpenLDAP's method to link entities- find all other entities that mention this user/group as a member in their schema. But AD code has always relied on 'memberof' attribute of the group/user itself to find what all groups it belongs to. So if AD relies more on 'memberof', this query will be different for AD vs OpenLDAP. Can we make sure that?

Reply:
What we need to ensure is group schema in AD will have 'member' attribute - since we have relied on memberOf so far

Reply:
Looking at this [answer](https://stackoverflow.com/questions/22003134/is-there-an-ldap-standard-group-membership-attribute-for-users), and the [RFC 4519 spec](https://tools.ietf.org/html/rfc4519#section-2.17) it refers, LDAP originally has `member` attribute for all implementations. But `memberOf` is something specifically for AD. So I think we're good for the nested group pull up